cxx03 test breaks cmake builds requiring recent C++

Importing Google Benchmark via FetchContent and loading it into a CMake project with a project-wide minimum standard version above 98 fails to compile in test/cxx03_test.cc

Example CMakeLists.txt for repro:

```
cmake_minimum_required(VERSION 3.14)
project(issue933)
enable_testing()

include(FetchContent)

## Project-wide setup
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED YES)
set(CMAKE_CXX_EXTENSIONS NO)

# Externally provided libraries
FetchContent_Declare(googletest
        GIT_REPOSITORY https://github.com/google/googletest.git
        GIT_TAG v1.10.x)

FetchContent_Declare(googlebenchmark
        GIT_REPOSITORY https://github.com/google/benchmark.git
        GIT_TAG master) # need master for benchmark::benchmark

FetchContent_MakeAvailable(
        googletest
        googlebenchmark)
```

Then try building:
```
$ mkdir build
$ cd build
$ cmake -DCMAKE_BUILD_TYPE=Debug ..
$ cmake --build .
FAILED: _deps/googlebenchmark-build/test/CMakeFiles/cxx03_test.dir/cxx03_test.cc.o 
/usr/bin/c++  -DHAVE_POSIX_REGEX -DHAVE_STD_REGEX -DHAVE_STEADY_CLOCK -I_deps/googlebenchmark-src/include -I_deps/googlebenchmark-src/src/../include -std=c++11  -Wall  -Wextra  -Wshadow  -pedantic  -pedantic-errors  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing -g   -std=c++03 -std=c++14 -MD -MT _deps/googlebenchmark-build/test/CMakeFiles/cxx03_test.dir/cxx03_test.cc.o -MF _deps/googlebenchmark-build/test/CMakeFiles/cxx03_test.dir/cxx03_test.cc.o.d -o _deps/googlebenchmark-build/test/CMakeFiles/cxx03_test.dir/cxx03_test.cc.o -c _deps/googlebenchmark-src/test/cxx03_test.cc
_deps/googlebenchmark-src/test/cxx03_test.cc:8:2: error: #error C++11 or greater detected. Should be C++03.
    8 | #error C++11 or greater detected. Should be C++03.
      |  ^~~~~
_deps/googlebenchmark-src/test/cxx03_test.cc:12:2: error: #error C++11 or greater detected by the library. BENCHMARK_HAS_CXX11 is defined.
   12 | #error C++11 or greater detected by the library. BENCHMARK_HAS_CXX11 is defined.
      |  ^~~~~
```

This can be patched by replacing line 151 of test/CMakeLists.txt:
```
151c151
<       COMPILE_FLAGS "-std=c++03")
---
>       CXX_STANDARD 98)
```

This should work across all supported compilers and CMake doesn't bother to draw a distinction between 98 and 03. See docs for [gcc](https://gcc.gnu.org/onlinedocs/gcc/Standards.html) (only treats warnings differently with `-pedantic`), [clang](https://clang.llvm.org/cxx_status.html#cxx98) (no distinction), [MSVC](https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=vs-2015) (doesn't support the flag), and [Intel](https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-std-qstd) (doesn't support the flag).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ foreach(p
     CMP0056 # export EXE_LINKER_FLAGS to try_run
     CMP0057 # Support no if() IN_LIST operator
     CMP0063 # Honor visibility properties for all targets
+    CMP0077 # Allow option() overrides in importing projects
     )
   if(POLICY ${p})
     cmake_policy(SET ${p} NEW)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -148,7 +148,8 @@ if (BENCHMARK_HAS_CXX03_FLAG)
   compile_benchmark_test(cxx03_test)
   set_target_properties(cxx03_test
       PROPERTIES
-      COMPILE_FLAGS "-std=c++03")
+      CXX_STANDARD 98
+      CXX_STANDARD_REQUIRED YES)
   # libstdc++ provides different definitions within <map> between dialects. When
   # LTO is enabled and -Werror is specified GCC diagnoses this ODR violation
   # causing the test to fail to compile. To prevent this we explicitly disable

